src/download.js

```diff
@@ -1,124 +1,98 @@
-
-			var tempAssetPath = assetPath + '.tmp';
-			onMiss(tempAssetPath, function (err) {
-				if (err) { return cb(err); }
-
-				fs.rename(tempAssetPath, assetPath, function (err) {
-					if (err) { return cb(err); }
-
-					cb(null, assetPath);
-				});
-			});
-		});
-	});
-}
-
 function download(opts, cb) {
-	var repo = opts.repo || 'atom/electron';
-	var github = new GitHub({ repo: repo, token: opts.token });
+	let bar;
 
 	if (!opts.version) {
 		return cb(new Error('Missing version'));
 	}
 
-	var platform = opts.platform;
-	if (!platform) {
+	if (!opts.platform) {
 		return cb(new Error('Missing platform'));
 	}
 
-	var arch = opts.arch;
-
+	let arch = opts.arch;
 	if (!arch) {
-		switch (platform) {
+		switch (opts.platform) {
 			case 'darwin': arch = 'x64'; break;
 			case 'win32': arch = 'ia32'; break;
 			case 'linux': arch = 'ia32'; break;
 		}
 	}
 
-	var version = 'v' + opts.version;
-	var assetName = [opts.assetName, version, platform, arch].join('-') + '.zip';
-
-	function download(assetPath, cb) {
-		github.getReleases({ tag_name: version }, function (err, releases) {
-			if (err) { return cb(err); }
-
-			var release = releases[0];
-
-			if (!release) {
-				return cb(new Error('No release ' + opts.version + ' found'));
-			}
-
-			var asset = release.assets.filter(function (asset) {
-				return asset.name === assetName;
-			})[0];
-
-			if (!asset) {
-				return cb(new Error('No asset for version ' + opts.version + ', platform ' + platform + ' and arch ' + arch + ' found'));
-			}
-
-			github.downloadAsset(asset, function (err, istream) {
-				if (err) {
-					console.error('Error creating download stream to download ' + asset.name);
-					return cb(err);
-				}
-
-				if (process.stdout.isTTY && !opts.quiet) {
-					var bar = new ProgressBar('↓ ' + asset.name + ' [:bar] :percent', {
-						total: asset.size,
-						width: 20
-					});
-
-					istream.on('data', function (chunk) { bar.tick(chunk.length); });
-				} else {
-					console.log('Downloading ' + asset.name + '...');
-				}
-
-				var ostream = fs.createWriteStream(assetPath);
-				istream.pipe(ostream);
-				istream.on('error', function (err) {
-					console.error('Error in input stream while downloading ' + asset.name);
-					cb(err);
-				});
-				ostream.on('error', function (err) {
-					console.error('Error in output stream while downloading ' + asset.name);
-					cb(err);
-				});
-				ostream.on('close', function () {
-					console.log('Downloaded ' + asset.name);
-					cb();
-				});
+	const artifactName = opts.assetName ? opts.assetName : 'electron'
+
+	const downloadOpts = {
+		version: opts.version,
+		platform: opts.platform,
+		arch,
+		artifactName,
+		token: opts.token,
+		downloadOptions: {
+			getProgressCallback: (progress) => {
+				if (bar) bar.update(progress.percent);
+			},
+		}
+	};
+
+	let assetToDownload = artifactName;
+	if (opts.repo) {
+		assetToDownload = `electron-${opts.version}-${opts.platform}-${arch}.zip`;
```

Comment:
Its not always electron, there is also ffmpeg. I tried push a change to the branch but permission was denied, can you make this changel.

```patch
diff --git a/src/download.js b/src/download.js
index fbfa2f3..728b4e8 100644
--- a/src/download.js
+++ b/src/download.js
@@ -46,13 +46,8 @@ function download(opts, cb) {
                }
        };
 
-       let assetToDownload = artifactName;
-       if (opts.repo) {
-               assetToDownload = `electron-${opts.version}-${opts.platform}-${arch}.zip`;
-       }
-
        bar = new ProgressBar(
-               `Downloading ${assetToDownload}: [:bar] :percent ETA: :eta seconds `,
+               `Downloading ${artifactName}: [:bar] :percent ETA: :eta seconds `,
                {
                        curr: 0,
                        total: 100,
```